packages/@aws-cdk/aws-lambda/lib/edge-function.ts

```diff
@@ -0,0 +1,266 @@
+import * as path from 'path';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as ssm from '@aws-cdk/aws-ssm';
+import {
+  BootstraplessSynthesizer, Construct as CoreConstruct, ConstructNode,
+  CustomResource, CustomResourceProvider, CustomResourceProviderRuntime,
+  DefaultStackSynthesizer, IStackSynthesizer, Resource, Stack, StackProps, Stage, Token,
+} from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { Alias, AliasOptions } from './alias';
+import { EventInvokeConfigOptions } from './event-invoke-config';
+import { IEventSource } from './event-source';
+import { EventSourceMapping, EventSourceMappingOptions } from './event-source-mapping';
+import { Function, FunctionProps } from './function';
+import { IFunction } from './function-base';
+import { calculateFunctionHash } from './function-hash';
+import { extractQualifierFromArn, IVersion } from './lambda-version';
+import { Permission } from './permission';
+
+/**
+ * Properties for creating a Lambda@Edge function
+ * @experimental
+ */
+export interface EdgeFunctionExperimentalProps extends FunctionProps { }
+
+/**
+ * A Lambda@Edge function.
+ *
+ * Convenience resource for requesting a Lambda function in the 'us-east-1' region for use with Lambda@Edge.
+ * Implements several restrictions enforced by Lambda@Edge.
+ *
+ * @resource AWS::Lambda::Function
+ * @experimental
+ */
+export class EdgeFunctionExperimental extends Resource implements IVersion {
```

Comment:
Following up on our conversation, let's move this to the cloudfront module.

Reply:
Done!